conans/test/functional/toolchains/cmake/test_file_api.py

```diff
@@ -103,8 +105,13 @@ class Elogio(ConanFile):
             generators = "CMakeDeps", "CMakeToolchain"
 
             def build(self):
+                file_api = CMakeFileAPI(self)
+                file_api.query(CMakeFileAPI.CODEMODELV2)
                 cmake = CMake(self)
                 cmake.configure()
+                reply = file_api.reply(CMakeFileAPI.CODEMODELV2)
```

Comment:
This part of the test is failing, as the system doesn't know how to parse executables

Reply:
I don't understand what do you mean...
I don't parse any executables, I only parse .json files, which are output of CMake File API
moreover, tests are green: https://ci.conan.io/job/ConanTestSuite/job/PR-9005/19/display/redirect

Reply:
I think I see now: `Exception: don't know how to convert use_decoder.exe`
I will fix that error, not a bit deal
however, I don't get why did you put file api here? this block doesn't have any `package_info`, it's just a consumer project, so it's basically useless here

Reply:
But it could become a package at anytime, and I want to use the feature to parse the targets and automatically get me the CppPackage and the package_info(). The feature cannot crash when building executables.

Reply:
yes, I agree that it cannot crash, I already addressed that in https://github.com/conan-io/conan/pull/9005/commits/f35b6c430ec19798ec195830fc561be548ab3b38
the question, should we leave that code there if it has no side effect?